apps/passport-client/src/registerServiceWorker.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Installs a service worker which caches application code and
+ * various artifacts needed by the application, so that the website
+ * works offline, and so that it loads fast.
+ *
+ * The service worker is invalidated each time there is a production
+ * deploy.
+ *
+ * The service worker is not installed in development mode, so that
+ * its caching of application code does not interfere with quick
+ * iteration loops.
+ */
+export async function registerServiceWorker() {
+  if (process.env.NODE_ENV === "development") {
+    console.log(
+      `[SERVICE_WORKER] not registering service worker in development mode`
```

Comment:
this is a bit scary.

can we test thoroughly locally?

if we run `NODE_ENV=production ...` to test locally, and it installs a service worker, will `yarn run dev` still work afterward?